### ISSUE TITLE
[BUG] WhatsApp database not found.

### ISSUE THREAD

**Issue by luxxed** [KnugiHK/WhatsApp-Chat-Exporter#125]
- WhatsApp version: 2.24.24.83
- OS: [Android/iOS] - iOS 18.2
- Platform: MacOS and Windows

WhatsApp database not found when attempting to export. 
Non encrypted backup. 
Backing up to iCloud inside WhatsApp as well. >5gb... 

I've tried both:

```
wtsexporter -i -b ~"/Library/Application Support/MobileSync/Backup/00008110-000A1C4A1E82801E"
```

```
wtsexporter -i -b ~/Library/Application\ Support/MobileSync/Backup/00008110-000A1C4A1E82801E
```

Tried with sudo etc. 
Able to navigate to the backup directory with Manifest.db
Unable to locate the file `7c7fba66680ef796b916b067077cc246adacf01d` in the 7c or backup folders... 

Any idea what's going on? @KnugiHK
Really don't want to sign out of my first phone's WhatsApp before this is sorted and I was planning on wiping the WhatsApp before starting fresh on the 2nd phone.. 

Cheers.

**Comment by KnugiHK** [comment]
Hi, does the `7c7fba66680ef796b916b067077cc246adacf01d` file appear in your Manifest.db?

**Comment by luxxed** [comment]
Hi, no it doesn't unfortunately. 
I also don't have a `AppDomainGroup-group.net.whatsapp.WhatsApp.shared-ChatStorage.sqlite`

<img width="1926" alt="Screenshot 2024-12-23 at 6 48 58 am" src="https://github.com/user-attachments/assets/0ee0f18b-893d-45f0-b282-fc749c5521e3" />

**Comment by luxxed** [comment]
Using an encrypted backup just to check
`Decrypting WhatsApp database...Essential WhatsApp files are missing from the iOS backup.`

What am I doing wrong?

**Comment by KnugiHK** [comment]
I haven't had time to review backups from iOS 18.2 yet, but could it be related to end-to-end encrypted WhatsApp backups or iCloud's Advanced Data Protection?

**Comment by luxxed** [comment]
Do I need to turn them off? I thought a computer backup wasn't related to the iCloud backup or WhatsApp backups. I'll give it a try nonetheless and see what happens. 

Let me know if you end up testing 18.2 backups and I'll report back too. 
Cheers and Merry Christmas

**Comment by KnugiHK** [comment]
I tested on an iPhone running iOS 18.2 with WhatsApp version 2.24.25.93.

It turned out that the file was correctly displayed in the Manifest.db file when end-to-end encryption was disabled, but it was not included when end-to-end encryption was enabled. It seems that WhatsApp developers deliberately excluded certain files from being backed up by computer.

PS. the end-to-end encryption I mentioned is the one provided by WhatsApp
![image](https://github.com/user-attachments/assets/525eba02-5649-4ba1-887b-1bb85da0938d)

**Comment by varenc** [comment]
running into this problem as well with iOS 18.2 and an encrypted iOS backup on my mac.  Does disabling E2EE in the WhatsApp "Chat backup" settings and then re-doing my iTunes backup seem likely to fix?

**Comment by varenc** [comment]
yes, it does! I can confirm that turning of E2EE backup in the Whatsapp Chat Backup settings does indeed seem to cause it to include more info in the iTunes encrypted backup.  Getting some warnings like this when running it:
```
WARN: decrypted 2678581 bytes of 'AppDomainGroup-group.net.whatsapp.WhatsApp.shared/BackedUpKeyValue.sqlite', expected 3604382 bytes!
WARN: decrypted 494618 bytes of 'AppDomainGroup-group.net.whatsapp.WhatsApp.shared/Biz/Biz.sqlite', expected 528262 bytes!
```
but other than that, it all seems to work!

**Comment by KnugiHK** [comment]
Thank you for confirming the issue! I believe the best approach would be to display a warning and suggest that users disable the E2EE backup.

**Comment by KnugiHK** [comment]
Released in 0.12.0.
